pkg/yurthub/filter/forwardkubesvctraffic/filter.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2024 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package forwardkubesvctraffic
+
+import (
+	"strconv"
+
+	discovery "k8s.io/api/discovery/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+	utilnet "k8s.io/utils/net"
+
+	"github.com/openyurtio/openyurt/pkg/yurthub/filter"
+	"github.com/openyurtio/openyurt/pkg/yurthub/filter/base"
+)
+
+const (
+	// FilterName filter is used to mutate the default/kubernetes endpointslices
+	// in order to make pods on edge nodes can access kube-apiserver directly by default/kubernetes service.
+	FilterName = "forwardkubesvctraffic"
+
+	KubeSVCNamespace = "default"
+	KubeSVCName      = "kubernetes"
+	KubeSVCPortName  = "https"
+)
+
+// Register registers a filter
+func Register(filters *base.Filters) {
+	filters.Register(FilterName, func() (filter.ObjectFilter, error) {
+		return NewForwardKubeSVCTrafficFilter()
+	})
+}
+
+func NewForwardKubeSVCTrafficFilter() (*forwardKubeSVCTrafficFilter, error) {
+	return &forwardKubeSVCTrafficFilter{addressType: discovery.AddressTypeIPv4}, nil
+}
+
+type forwardKubeSVCTrafficFilter struct {
+	addressType discovery.AddressType
+	host        string
+	port        int32
+}
+
+func (fkst *forwardKubeSVCTrafficFilter) Name() string {
+	return FilterName
+}
+
+func (fkst *forwardKubeSVCTrafficFilter) SupportedResourceAndVerbs() map[string]sets.String {
+	return map[string]sets.String{
+		"endpointslices": sets.NewString("list", "watch"),
```

Comment:
Does it need to support `endpoints`?

Reply:
I think there is no need to support endpoints resource, because EndpointSlice is stable from K8s v1.21, and most  users are currently using K8s versions greater than v1.22.

Reply:
But there're still other filters that support endpoints, like serviceTopologyFilter. It's better to make them all support `endpoints` or not.

BTW, I think we can support the endpoints, since currently it does not take us much effort.

Reply:
It is different with serviceTopologyFilter, because some other components like `nginx-ingress-controller` that are using endpoints resource also need servicetopology filter, but forwardkubesvctraffic filter is only by kube-proxy, so there is no need to deal with endpoints.